File: Entity/Cliente.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entity
{
    public class Cliente : Persona
    {
        [Column(TypeName = "varchar(4)")]
        public string IdCliente { get; set; }

        [NotMapped]
        public IList<Reserva> Reservas { get; set; }

        [NotMapped]
        public Habitacion Habitacion { get; set; }

        [Column(TypeName = "varchar(4)")]
        public string IdHabitacion { get; set; }

        [Column(TypeName = "varchar(12)")]
        public string Ppal { get; set; }
    }
}

File: proyecto/ClientApp/src/app/hotel/models/cliente.ts
import { Factura } from "./factura";
import { Habitacion } from "./habitacion";
import { Persona } from "./persona";

export class Cliente extends Persona{
    idcliente: string;
    factura: Factura;
    habitacion: Habitacion;
    idhabitacion:string;
    ppal: string;
}
